packages/astro/src/integration/create-integration.ts

```diff
@@ -0,0 +1,112 @@
+
+          const buildImportPath =
+            mode === 'bundled' ? defaultBundledImportPath : defaultBundledImportPath.replace('/bundled', '');
+
+          // Set params as envs do backend code has access to them
+          updateConfig({
+            vite: {
+              define: {
+                ...buildEnvVarFromOption(signInUrl, 'PUBLIC_ASTRO_APP_CLERK_SIGN_IN_URL'),
+                ...buildEnvVarFromOption(signUpUrl, 'PUBLIC_ASTRO_APP_CLERK_SIGN_UP_URL'),
+                ...buildEnvVarFromOption(isSatellite, 'PUBLIC_ASTRO_APP_CLERK_IS_SATELLITE'),
+                ...buildEnvVarFromOption(proxyUrl, 'PUBLIC_ASTRO_APP_CLERK_PROXY_URL'),
+                ...buildEnvVarFromOption(domain, 'PUBLIC_ASTRO_APP_CLERK_DOMAIN'),
+                ...buildEnvVarFromOption(domain, 'PUBLIC_ASTRO_APP_CLERK_DOMAIN'),
+                ...buildEnvVarFromOption(domain, 'PUBLIC_ASTRO_APP_CLERK_DOMAIN'),
+                ...buildEnvVarFromOption(clerkJSUrl, 'PUBLIC_ASTRO_APP_CLERK_JS_URL'),
+                ...buildEnvVarFromOption(clerkJSVariant, 'PUBLIC_ASTRO_APP_CLERK_JS_VARIANT'),
+                ...buildEnvVarFromOption(clerkJSVersion, 'PUBLIC_ASTRO_APP_CLERK_JS_VERSION'),
+                __HOTLOAD__: mode === 'hotload',
+              },
+
+              ssr: {
+                external: ['node:async_hooks'],
+              },
+
+              // We need this for top-level await
+              optimizeDeps: {
+                esbuildOptions: {
+                  target: 'es2022',
+                },
+              },
+              build: {
+                target: 'es2022',
+              },
+            },
+          });
+
+          /**
+           * The above script will run before client frameworks like React hydrate.
+           * This makes sure that we have initialized a Clerk instance and populated stores in order to avoid hydration issues
+           */
+          injectScript(
+            'before-hydration',
+            `
+            ${command === 'dev' ? `console.log('${packageName}',"Initialize Clerk: before-hydration")` : ''}
+            import { runInjectionScript } from "${buildImportPath}";
+            await runInjectionScript(${JSON.stringify(params)});`,
+          );
+
+          /**
+           * The above script only executes if a client framework like React needs to hydrate.
+           * We need to run the same script again for each page in order to initialize Clerk even if no UI framework is used in the client
+           * If no UI framework is used in the client, the above script with `before-hydration` will never run
+           */
+
+          injectScript(
+            'page',
+            `
+            ${command === 'dev' ? `console.log("${packageName}","Initialize Clerk: page")` : ''}
+            import { runInjectionScript } from "${buildImportPath}";
+            await runInjectionScript(${JSON.stringify(params)});`,
+          );
+        },
```

Comment:
Will these two both run if a UI framework is present? Is that problematic?

Reply:
There will be no problem, because `runInjectionScript` will only execute clerk-js initialization once

Reply:
I will write more comments about why we need to have both.